extension/arquillian-drone-browserstack-extension/src/main/java/org/arquillian/drone/browserstack/extension/local/BrowserStackLocalRunner.java

```diff
@@ -0,0 +1,257 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016, Red Hat Middleware LLC, and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.arquillian.drone.browserstack.extension.local;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import org.arquillian.spacelift.Spacelift;
+import org.arquillian.spacelift.task.archive.UnzipTool;
+import org.arquillian.spacelift.task.net.DownloadTool;
+
+/**
+ * Is responsible for starting a BrowserStackLocal binary
+ *
+ * @author <a href="mailto:mjobanek@redhat.com">Matous Jobanek</a>
+ */
+public class BrowserStackLocalRunner {
+
+    private static Logger log = Logger.getLogger(BrowserStackLocalRunner.class.getName());
+
+    private static BrowserStackLocalRunner browserStackLocalRunner = null;
+
+    private final File browserStackLocalDirectory = new File("target" + File.separator + "browserstacklocal");
+    private final File browserStackLocalFile =
+        new File(browserStackLocalDirectory.getPath() + File.separator + "BrowserStackLocal" + (PlatformUtils
+            .isWindows() ? ".exe" : ""));
+    private final String basicUrl = "https://www.browserstack.com/browserstack-local/";
+
+    private final CountDownLatch countDownLatch = new CountDownLatch(1);
+    private Process process = null;
+
+    private BrowserStackLocalRunner() {
+    }
+
+    /**
+     * Returns an instance of BrowserStackLocalRunner. If there has been already created, returns this one, otherwise
+     * creates and returns a new one - behaves like singleton
+     *
+     * @return An instance of BrowserStackLocalRunner
+     */
+    public static BrowserStackLocalRunner createBrowserStackLocalInstance() {
```

Comment:
I would keep it as it is because of lazy initialization. This class is created only if the user has appropriate property set to true, otherwise it won't be created.